Sources/NIOCore/ScheduledOnCurrentEventLoop.swift

```diff
@@ -0,0 +1,36 @@
+public struct ScheduledOnCurrentEventLoop<T> {
+    public let wrapped: Scheduled<T>
+    
+    public var futureResult: CurrentEventLoopFuture<T> {
+        wrapped.futureResult.iKnowIAmOnTheEventLoopOfThisFuture()
+    }
+    
+    @inlinable init(_ scheduled: Scheduled<T>) {
+        self.wrapped = scheduled
+    }
+    
+    @inlinable
+    public init(promise: CurrentEventLoopPromise<T>, cancellationTask: @escaping () -> Void) {
+        wrapped = .init(promise: promise.wrapped, cancellationTask: cancellationTask)
+    }
+    
+    @inlinable
+    public func cancel() {
+        wrapped.cancel()
+    }
+}
+
+#if swift(>=5.6)
+@available(*, unavailable)
+extension ScheduledOnCurrentEventLoop: Sendable {}
+#endif
+
+extension Scheduled {
+    @inlinable public func iKnowIAmOnTheEventLoopOfThisScheduled(
```

Comment:
The name is trying a bit too hard to be funny -- can we have something that explains what it is doing?

Specifically this is `assumeOnScheduledEventLoop()` or something; `assume...` sounds good to me as a "assume or crash", so that could be adopted as general wording.